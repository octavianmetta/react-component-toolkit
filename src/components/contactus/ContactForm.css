/* Gaya untuk input */
input[type="text"],
input[type="email"],
input[type="tel"],
textarea {
  color: #000;
  font-size: 16px;
  font-family: Arial;
  font-weight: 400;
  line-height: 25px;
  word-wrap: break-word;
  padding: 10px;
  display: block;
  width: 100%;
  max-width: 100%; /* Atur lebar maksimum sesuai dengan lebar form */
  box-sizing: border-box; /* Menghitung padding dan border dalam lebar total elemen */

  margin-bottom: 15px;
  border: 1px solid #dadee3;
  border-radius: 8px;
}

/* Gaya untuk placeholder */
input[type="text"]::placeholder,
input[type="email"]::placeholder,
input[type="tel"]::placeholder,
textarea::placeholder {
  color: #dadee3;
}

/* Gaya untuk tombol kirim */
input[type="submit"] {
  padding: 15px 20px;
  border: none;
  background: #09101d;
  color: white;
  font-size: 16px;
  font-family: Arial;
  font-weight: 400;
  line-height: 25px;
  border-radius: 8px;
  cursor: pointer;
}

input[type="submit"]:hover {
  background: #0e162e;
}

textarea {
  resize: vertical;
}

h4 {
  color: #c2141c;
  font-size: 28px;
  font-family: Arial;
  font-weight: 700;
  line-height: 32px;
  word-wrap: break-word;
}

/* Gaya untuk menempatkan tombol submit di kanan */
form {
  text-align: right; /* Mengatur posisi konten dalam form menjadi ke kanan */

  outline: 1px solid red; /* Menampilkan outline pada elemen form */
  max-width: 500px; /* Atur lebar maksimum form */
}
