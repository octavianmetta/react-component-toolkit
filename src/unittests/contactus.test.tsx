// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { faker } from "@faker-js/faker";
import ContactForm from "../components/contactus/contactus";

const componentId: string = "ComponentUnderTest";
let faketitle: string = "";
let fakecontactEmail: string = "";

describe("Testing component : <contactus />", () => {
  const componentId = "ComponentUnderTest";

  beforeAll(() => {});

  afterAll(() => {});

  beforeEach(() => {
    faketitle = faker.lorem.words(4);
    fakecontactEmail = faker.internet.email();
  });

  afterEach(() => {
    faketitle = "";
    fakecontactEmail = "";
  });

  test("<contactus /> should render with id", async () => {
    // Create our component with our generated id
    const testRenderer = renderer.create(
      <ContactForm id={componentId} title="" contactEmail="" />
    );

    // Use the created renderer to convert to json and then check it matches our expected snapshot
    const jsonSnapshot = testRenderer.toJSON();

    // Check that our rendered component does have the exact randomly generated componentid for this test
    expect(jsonSnapshot).toHaveProperty("props.id", componentId);

    // Match that our rendered component matches our snapshot of the component and ignore the random id
    expect(jsonSnapshot).toMatchSnapshot({
      props: { id: expect.any(String) },
    });
  });
});
