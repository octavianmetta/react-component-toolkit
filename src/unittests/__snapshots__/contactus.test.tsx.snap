// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component : <contactus /> <contactus /> should render with id 1`] = `
{
  "children": null,
  "props": {
    "className": "sc-bcXHqe ileYXv",
    "data-testid": "contactusTestId",
    "id": Any<String>,
  },
  "type": "h1",
}
`;
